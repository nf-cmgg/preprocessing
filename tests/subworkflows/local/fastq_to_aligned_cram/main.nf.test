nextflow_workflow {

    name "Test Workflow FASTQ_TO_CRAM"
    script "subworkflows/local/fastq_to_aligned_cram/main.nf"
    workflow "FASTQ_TO_CRAM"

    tag "subworkflows"
    tag "subworkflows/local"
    tag "subworkflows/local/fastq_to_aligned_cram"

    test("fastq to cram - bwa - bamsormadup") {
        when {
            workflow {
                """
                // [meta, [fq_1,fq_2], aligner_index]
                input[0] = Channel.of([
                    [
                        id:'test',
                        samplename:'test',
                        single_end:false,
                        genome: [
                            fasta: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna",
                            fai: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna.fai"
                        ]
                    ], // meta map
                    [
                        file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/fastq/sample1_R1.fastq.gz", checkIfExists: true),
                        file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/fastq/sample1_R2.fastq.gz", checkIfExists: true)
                    ],
                    "bwamem", // aligner
                    file("s3://test-data/genomics/homo_sapiens/genome/bwa/", checkIfExists: true),
                    file("s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna")
                ])
                // markdup
                input[1] = "bamsormadup"
                """
            }
        }

        then {
            assertAll(
                {
                    assert workflow.success
                    assert snapshot(
                        file(workflow.out.cram_crai[0][1]).name,
                        file(workflow.out.cram_crai[0][2]).name,
                        workflow.out.multiqc_files,
                        workflow.out.versions
                    ).match()
                }
            )
        }
    }

    test("fastq to cram - bwa - samtools sormadup") {
        when {
            workflow {
                """
                // [meta, [fq_1,fq_2], aligner, index, fasta]
                input[0] = Channel.of([
                    [
                        id:'test',
                        samplename:'test',
                        single_end:false,
                        genome: [
                            fasta: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna",
                            fai: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna.fai"
                        ]
                    ], // meta map
                    [
                        file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/fastq/sample1_R1.fastq.gz", checkIfExists: true),
                        file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/fastq/sample1_R2.fastq.gz", checkIfExists: true)
                    ],
                    "bwamem", // aligner
                    file("s3://test-data/genomics/homo_sapiens/genome/bwa/", checkIfExists: true),
                    file("s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna")
                ])
                // markdup
                input[1] = "samtools"
                """
            }
        }

        then {
            assertAll(
                {
                    assert workflow.success
                    assert snapshot(
                        file(workflow.out.cram_crai[0][1]).name,
                        file(workflow.out.cram_crai[0][2]).name,
                        workflow.out.multiqc_files,
                        workflow.out.versions
                    ).match()
                }
            )
        }
    }

    test("fastq to cram - bwa - samtools sort") {
        when {
            workflow {
                """
                // [meta, [fq_1,fq_2], aligner, index, fasta]
                input[0] = Channel.of([
                    [
                        id:'test',
                        samplename:'test',
                        single_end:false,
                        genome: [
                            fasta: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna",
                            fai: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna.fai"
                        ]
                    ], // meta map
                    [
                        file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/fastq/sample1_R1.fastq.gz", checkIfExists: true),
                        file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/fastq/sample1_R2.fastq.gz", checkIfExists: true)
                    ],
                    "bwamem", // aligner
                    file("s3://test-data/genomics/homo_sapiens/genome/bwa/", checkIfExists: true),
                    file("s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna")
                ])
                // markdup
                input[1] = "false"
                """
            }
        }

        then {
            assertAll(
                {
                    assert workflow.success
                    assert snapshot(
                        file(workflow.out.cram_crai[0][1]).name,
                        file(workflow.out.cram_crai[0][2]).name,
                        workflow.out.multiqc_files,
                        workflow.out.versions
                    ).match()
                }
            )
        }
    }

    test("fastq to cram - stub") {
        options: "-stub"
        when {
            workflow {
                """
                // [meta, [fq_1,fq_2], aligner, index, fasta]
                input[0] = Channel.of([
                    [
                        id:'test',
                        samplename:'test',
                        single_end:false,
                        genome: [
                            fasta: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna",
                            fai: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna.fai"
                        ]
                    ], // meta map
                    [
                        file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/fastq/sample1_R1.fastq.gz", checkIfExists: true),
                        file("https://github.com/nf-cmgg/test-datasets/raw/preprocessing/data/genomics/homo_sapiens/illumina/fastq/sample1_R2.fastq.gz", checkIfExists: true)
                    ],
                    "bwamem", // aligner
                    file("s3://test-data/genomics/homo_sapiens/genome/bwa/", checkIfExists: true),
                    file("s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna")
                ])
                // markdup
                input[1] = "false"
                """
            }
        }

        then {
            assertAll(
                {
                    assert workflow.success
                    assert snapshot(
                        file(workflow.out.cram_crai[0][1]).name,
                        file(workflow.out.cram_crai[0][2]).name,
                        workflow.out.multiqc_files,
                        workflow.out.versions
                    ).match()
                }
            )
        }
    }
}
