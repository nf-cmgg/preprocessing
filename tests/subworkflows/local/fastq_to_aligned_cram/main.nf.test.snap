{
    "fastq to cram - stub": {
        "content": [
            "test.merged.cram",
            "test.merged.cram.crai",
            [
                
            ],
            [
                "versions.yml:md5,1628f3785f7e7fd46344418b00926ff3",
                "versions.yml:md5,bbf4851478b2a3c0f4d48343af3b647d"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-25T20:25:41.445503"
    },
    "fastq to cram - bwa - bamsormadup": {
        "content": [
            "test.cram",
            "test.cram.crai",
            [
                "test.merged.metrics.txt:md5,4ee20f12cb4d6077479a08310c8f6c70"
            ],
            [
                "versions.yml:md5,bbf4851478b2a3c0f4d48343af3b647d",
                "versions.yml:md5,d8544811f6b511ef45e9c3547430a30d"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-25T20:21:05.679819"
    },
    "fastq to cram - bwa - samtools sormadup": {
        "content": [
            "test.merged.cram",
            "test.merged.cram.crai",
            [
                "test.merged.metrics:md5,934298c4be265f81fdc3e39b303d811b"
            ],
            [
                "versions.yml:md5,163edee2f4f70839ca5012263163ffd1",
                "versions.yml:md5,bbf4851478b2a3c0f4d48343af3b647d"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-25T20:22:35.020168"
    },
    "fastq to cram - bwa - samtools sort": {
        "content": [
            "test.merged.cram",
            "test.merged.cram.crai",
            [
                
            ],
            [
                "versions.yml:md5,1628f3785f7e7fd46344418b00926ff3",
                "versions.yml:md5,bbf4851478b2a3c0f4d48343af3b647d"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-25T20:24:07.689047"
    }
}